pkg/xray/watches.go

```diff
@@ -162,24 +200,46 @@ func packProjectResources(resources WatchProjectResources) []interface{} {
 		if len(res.RepoType) > 0 {
 			resourceMap["repo_type"] = res.RepoType
 		}
-		resourceMap["filter"] = packFilters(res.Filters)
+
+		resourceMap = packFilters(res.Filters, resourceMap)
+
 		list = append(list, resourceMap)
 	}
 
 	return list
 }
 
-func packFilters(filters []WatchFilter) []interface{} {
-	var l []interface{}
-	for _, f := range filters {
-		m := map[string]interface{}{
-			"type":  f.Type,
-			"value": f.Value,
+func packFilters(filters []WatchFilter, resources map[string]interface{}) (map[string]interface{}) {
+	var stringFilters []map[string]interface{}
+	var antFilters []map[string]interface{}
+
+	for _, filter := range filters {
+		var f map[string]interface{}
+
+		switch filter.Type {
+		case "regex", "package-type":
+			var value string
+			json.Unmarshal(filter.Value, &value)
```

Comment:
You need a default case

Reply:
You should do `json.Unmarshal(filter.Value,&value)` before the case statement

see if you can remove some of this redundancy

Reply:
I thought of that but we can't because the type for `value` is different for each `case`